testing/describes.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FakeCustomElements, FakeWindow} from './fake-dom';
+import {doNotLoadExternalResourcesInTest} from './iframe';
+import {
+  adopt,
+  installAmpdocServices,
+  installRuntimeServices,
+  registerForUnitTest,
+} from '../src/runtime';
+import {installDocService} from '../src/service/ampdoc-impl';
+import {installExtensionsService} from '../src/service/extensions-impl';
+import {resetScheduledElementForTesting} from '../src/custom-element';
+import * as sinon from 'sinon';
+
+/** Should have something in the name, otherwise nothing is shown. */
+const SUB = ' ';
+
+
+/** @type {number} */
+let iframeCount = 0;
+
+
+/**
+ * - ampdoc: "single", "shadow", "multi", "none".
+ *
+ * @typedef {{
+ *   runtimeOn: (boolean|undefined),
+ *   ampdoc: (string),
+ * }}
+ */
+export let AmpTestSpec;
+
+
+/**
+ * A test with a sandbox.
+ * @param {string} name
+ * @param {{
+ *   fakeClock: (boolean|undefined),
+ * }} spec
+ * @param {function({
+ *   sandbox: !Sandbox,
+ *   clock: (!FakeClock|undefined),
+ * })} fn
+ */
+export function sandboxed(name, spec, fn) {
```

Comment:
How about just making it a part of the realWinSpec then?
I just don't like that we need to repeat such a hiarachy in every test:
```
describes.sandboxed(()=>{
  describes.readWindow()=>
```